README.md

```diff
@@ -1 +1,130 @@
+- PUT data into a workspace
+- Auto-generate identity keys from a 4 character name
+- Search via querystring params and path prefixes
+
+## Dream API
+
+Basic URL format:
+
+```
+earthstar://+gardening.xxxxxxxxx/path/to/key.json
+```
+
+GET document
+
+```JavaScript
+const res = await fetch('earthstar://+gardening.xxxxx/example')
+
+// Get the timestamp for when it was last updated
+res.headers.get('mtime') === "1231231231231"
+// Get the author for this document
+res.headers.get('earthstar-author') === '@bbbb.xxxxxxxx'
+// Get the signature for this document, in case you want it 🤷
+res.headers.get('earthstar-signature') === 'xxxxxxx'
+
+// Get the content of the document back out.
+// If it's JSON, you should use `.json()`
+const content = await res.text()
+```
+
+GET an identity 'name' - Generate an identity
+
+```JavaScript
+// Specify `@XXXX` where `X` is a lowercase ascii character
+// This will generate and store a keypair for this identity
+const res = fetch('earthstar://@xmpl')
+
+// Get the generated address for your identity
+const {address} = await res.json()
+
+address === '@xmpl.xxxxxxx'
+```
+
+POST some data, sign with identity
+
+```JavaScript
+// It's good practice to use a file extension so that the proper mime type gets set
+const res = await fetch('earthstar://+gardening.xxxxx/example.json', {
+  method: 'post',
+  body: JSON.stringify({
+    hello: 'world'
+  }),
+  headers: {
+    // Use the identity address you got before to sign your update
+    "earthstar-author": "@xmpl.xxxxxx"
+  }
+})
+
+// Get the timestamp for when it was last updated
+res.headers.get('mtime') === "1231231231231"
+// Your author address
+res.headers.get('earthstar-author') === '@xmpl.xxxxxxxx'
+// Get the signature for this document, in case you want it 🤷
+res.headers.get('earthstar-signature') === 'xxxxxxx'
+```
+
+POST some data scoped to your identity (others cant update)
+
+```JavaScript
+// Note the `/~@xml,
+const res = await fetch('earthstar://+gardening.xxxxx/example/~@xmpl.xxxxx/profile.json', {
+  method: 'post',
+  body: JSON.stringify({
+    name: 'Example'
+  }),
+  headers: {
+    // Use the identity address you got before to sign your update
+    "earthstar-author": "@xmpl.xxxxxx"
+  }
+})
+```
+
+Search - A get request on a prefix with querystring parameters
+
+Search options can be placed in the queryString section of the URL, more info can be found in the [QueryOpts interface](https://github.com/earthstar-project/earthstar/blob/master/src/util/types.ts#L194).
+
+```JavaScript
+// If you use a `/` at the end of your path, it'll do a search for things inside that path
+// This is equivalent to the `path` parameter in QueryOpts
+const res = await fetch('earthstar://+gardening.xxxxx/example/')
+
+/ If you use a `*` at the end of your path, it'll do a search for paths tht start with that prefix
+// This is equivalent to the `pathPrefix` parameter in QueryOpts
+const res = await fetch('earthstar://+gardening.xxxxx/example*')
```

Comment:
This would work although I would usually want to do `/example/*` instead of `/example*`

Reply:
Actually, given your [previous comment](https://github.com/earthstar-project/earthstar-fetch/pull/1#discussion_r570590266) If it ends in a `/` then we don't need the `*`.

Reply:
Hmm true!

So we have 3 behaviors, how about this:

1. Ends in not-a-slash, it's an exact path, only get that one
2. Ends in '*', do path prefix query and concatenate the... content by default?
3. Ends in `/`, look for index.html or index.md or index.txt, if not found then do a path prefix query and render it as a list of paths that are links, like an old school webserver index page?

Probably some people will be surprised that there's a difference between `/foo` and `/foo/` since it doesn't always matter on some sites, but it's ok with me.

Also, sadly, `*` is a legal character in Earthstar paths... but it's not recommended to use it... I can remove it from the spec, nobody is using it for anything yet.

Reply:
Would a different character than `*` be better? Do we need to have a prefix query that isn't an entire path?

Reply:
Hmm, I could see it being useful for stuff like quadkeys actually. Denying `*` or using a different special character would be cool.

Reply:
Yeah sometimes we would want to do a prefix that's not an entire path segment.

Another example is that apps put version numbers on their data so they can change schemas later: `/todolist-v1/`.  You might want all versions: `/todolist-v*`

I'm happy to remove `*` from the allowed path characters.  Let me also make a list of non-path characters to think about...

Reply:
Thanks to my past self for writing notes about what characters are allowed and disallowed, and why:
https://github.com/earthstar-project/earthstar/blob/main/src/util/characters.ts#L37-L71

The only other available character that is URL-legal but not Earthstar-path-legal is `;`.  That doesn't feel like a prefix query to me, more like something for packing multiple Earthstar paths together into an URL, or something.

I'm going to deprecate `*` in Earthstar paths.  Let's use it in fetch, it seems intuitive to me.